File: solutions/88-musfik/AgeOfVillagers/Nation/ArabBedouin.cs
using System;
using System.Drawing;

namespace AgeOfVillagers
{
    public class ArabBedouin : INation
    {
        private readonly Color color = Color.LightGoldenrodYellow;
        private String nation_name = "Arab Bedouin";

        public ArabBedouin()
        {
            
        }

        public string nationName { get => nation_name; set => nation_name = value; }

        public House getHouse(Point point)
        {
            return new ArabHouse(point);
        }

        
        public string getNationName()
        {
            return nation_name;
        }
        

        public Color getTerrainColor()
        {
            return color;
        }

        public Tree getTree(Point point)
        {
            return new ArabTree(point);
        }

        public WaterResource getWaterResource(Point point)
        {
            return new NoWaterResource(point);
        }
    }
}

File: solutions/72-rafidhaque/src/ArabBedouin.java
import javafx.scene.Group;
import javafx.scene.paint.Color;
import javafx.scene.shape.Rectangle;

public class ArabBedouin implements INation {

    String nationName = "Arab Bedouin";
    Group drawSpace;
    DrawHouse house;
    DrawTree tree;
    DrawWater water;

    public ArabBedouin(Group drawSpace) {
        this.drawSpace = drawSpace;
        Rectangle fillColor = new Rectangle(0,0, 600, 400);
        fillColor.setFill(Color.LIGHTYELLOW);
        drawSpace.getChildren().add(fillColor);
        house = new ArabBedouinHouse();
        tree = new ArabBedouinTree();
        water = new NoWater( 0, 0);
        drawSpace.getChildren().addAll(house.draw());
        drawSpace.getChildren().addAll(tree.draw());
        drawSpace.getChildren().addAll(water.draw());
    }

    public String getNationName() {
        return nationName;
    }

    public DrawHouse getHouse() {
        return house;
    }

    public DrawTree getTree() {
        return tree;
    }

    public DrawWater getWater() {
        return water;
    }
}
